src/libs/auth.js

```diff
@@ -21,6 +21,27 @@ export const reloadAuthToken = () => {
   return getAuthInstance().currentUser.get().reloadAuthResponse().then(() => true, () => false)
 }
 
+export const hasBillingScope = () => {
+  return getAuthInstance().currentUser.get().hasGrantedScopes('https://www.googleapis.com/auth/cloud-billing')
+}
+
+/*
+ * Request Google Cloud Billing scope if necessary.
+ *
+ * NOTE: Requesting additional scopes may invoke a browser pop-up which the browser might block.
+ * If you use ensureBillingScope during page load and the pop-up is blocked, a rejected promise will
+ * be returned. In this case, you'll need to provide something for the user to deliberately click on
+ * and retry ensureBillingScope in reaction to the click.
+ */
+export const ensureBillingScope = async () => {
+  if (!hasBillingScope()) {
+    const options = new window.gapi.auth2.SigninOptionsBuilder({ 'scope': 'https://www.googleapis.com/auth/cloud-billing' })
+    await getAuthInstance().currentUser.get().grant(options)
+    // Wait 100ms before continuing to avoid errors due to delays in applying the new scope grant
+    await Utils.delay(100)
```

Comment:
This should resolve #1331, nice!

Reply:
Unfortunately, I don't think so. This serves the same role as the 250ms delay from your earlier PR (before the original origin of this branch). If we're still seeing the error after waiting 250ms then we'll probably need something more sophisticated here, like polling the auth instance until it says the scope is granted instead of just waiting for a fixed time interval.